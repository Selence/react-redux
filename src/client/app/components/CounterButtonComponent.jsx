import React from 'react';
import PropTypes from 'prop-types';

/**
 * @param {Object} actions
 * @returns {XML}
 */
export const CounterButtonComponent = ({actions}) => {

    const { increaseCounter } = actions;

    return (
        <div
            className="counter__button"
            onClick={increaseCounter}>
            Increase Counter +1
        </div>
    )

};

CounterButtonComponent.propTypes = {
    actions: PropTypes.shape({
        increaseCounter: PropTypes.func
    })
};

