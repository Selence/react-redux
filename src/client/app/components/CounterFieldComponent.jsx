import React from 'react';
import PropTypes from 'prop-types';

/**
 * @param {Number} counter
 * @returns {XML}
 */
export const CounterFieldComponent = ({
    counter
}) => {

    const renderHint = () => (
        <div className="hint">
            <p className="hint__text">Download <a href="https://github.com/zalmoxisus/redux-devtools-extension">redux-devtools-extension</a> to show the store</p>
        </div>
    );

    return (
        <div>
            <p>Counter:</p>
            <div className="counter__field">{counter}</div>
            {counter > 1 && renderHint()}
        </div>
    );
};

CounterFieldComponent.propTypes = {
    counter: PropTypes.number
};
