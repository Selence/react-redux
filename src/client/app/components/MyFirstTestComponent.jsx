import React from 'react';
import PropTypes from 'prop-types';

/**
 * @returns {XML}
 */
export const MyFirstTestComponent = ({
    counter
}) => {
    return (
        <div>
            uh...Yes I can!
        </div>
    );
};

MyFirstTestComponent.propTypes = {
    counter: PropTypes.number
};
