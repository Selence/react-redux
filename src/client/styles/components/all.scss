@import "counter";
@import "app";
@import "hint";