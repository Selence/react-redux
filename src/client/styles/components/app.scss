.app {
  width: 100%;

  @include element('text') {
    font-size: 25px;
  }
}