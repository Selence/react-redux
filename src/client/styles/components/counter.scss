.counter {
  width: 20%;

  @include element('button') {
    @include nonSelectable();

    background-color: lightgray;
    border: 1px solid lightgray;
    padding: 20px;
    text-align: center;
    width: 100%;

    &:hover {
      border-color: black;
    }

    &:active {
      background-color: white;
    }
  }

  @include element('field') {
    border: 1px dashed black;
    width: 10%;
  }
}