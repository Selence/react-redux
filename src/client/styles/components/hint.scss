.hint {
  border: 1px solid gray;
  margin-top: 10px;

  @include element('text') {
    padding: 15px;
  }
}