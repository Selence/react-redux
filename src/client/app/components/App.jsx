import React, { Component } from 'react';
import { MyFirstTestComponent } from './MyFirstTestComponent';

/**
 * @returns {XML}
 */
export class App extends Component {
    constructor(props) {
        super(props);
    }

    /**
     * @returns {XML}
     */
    render () {
        return (
            <div>
                <p>Hello React Project ... can you load ... ?</p>
                <MyFirstTestComponent />
            </div>
        );
    }
}