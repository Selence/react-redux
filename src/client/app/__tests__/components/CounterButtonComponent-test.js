import React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';
import {CounterButtonComponent} from '../../components/CounterButtonComponent';

describe('Component CounterButton', () => {
    let componentData,
        counterActionSpy;

    const renderComponent = () => (
        renderer.create(<CounterButtonComponent {...componentData}/>).toJSON()
    );

    const mountComponent = () => (
        mount(<CounterButtonComponent {...componentData}/>)
    );

    describe('when the component is rendered', () => {
        beforeEach(() => {
            counterActionSpy = jest.fn();

            componentData = {
                actions: {
                    increaseCounter: counterActionSpy
                }
            };
        });

        it('it math to snapshot', () => {
            expect(renderComponent()).toMatchSnapshot();
        });
    });

    describe('when the component is mounted', () => {
        let _component;

        beforeEach(() => {
            counterActionSpy = jest.fn();

            componentData = {
                actions: {
                    increaseCounter: counterActionSpy
                }
            };
        });

        describe('and the user has clicked the element to increase the counter', () => {
            beforeEach(() => {
                _component = mountComponent();
                _component.find('div').first().simulate('click');
            });

            it('the increaseCounter Action is called', () => {
                expect(counterActionSpy).toHaveBeenCalled();
            });
        })
    });
});