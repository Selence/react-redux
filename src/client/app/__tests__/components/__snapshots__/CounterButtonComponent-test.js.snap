// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component CounterButton when the component is rendered it math to snapshot 1`] = `
<div
  className="counter__button"
  onClick={[Function]}
>
  Increase Counter +1
</div>
`;
